app/screens/home/search/results/results.tsx

```diff
@@ -32,36 +47,81 @@ const AnimatedFlatList = Animated.createAnimatedComponent(FlatList);
 
 type Props = {
     searchValue: string;
-    selectedTab: 'messages' | 'files';
+    selectedTab: TabType;
     currentTimezone: string;
     isTimezoneEnabled: boolean;
     posts: PostModel[];
+    fileChannels: ChannelModel[];
     fileInfos: FileInfo[];
     scrollRef: React.RefObject<FlatList>;
     onScroll: (event: NativeSyntheticEvent<NativeScrollEvent>) => void;
     scrollPaddingTop: number;
     loading: boolean;
+    canDownloadFiles: boolean;
+    publicLinkEnabled: boolean;
 }
 
 const emptyList: FileInfo[] | Array<string | PostModel> = [];
 
-const SearchResults = ({
+const Results = ({
     currentTimezone,
     fileInfos,
     isTimezoneEnabled,
     posts,
+    fileChannels,
     searchValue,
     selectedTab,
     scrollRef,
     onScroll,
     scrollPaddingTop,
     loading,
+    canDownloadFiles,
+    publicLinkEnabled,
 }: Props) => {
     const theme = useTheme();
     const paddingTop = useMemo(() => ({paddingTop: scrollPaddingTop, flexGrow: 1}), [scrollPaddingTop]);
-
     const orderedPosts = useMemo(() => selectOrderedPosts(posts, 0, false, '', '', false, isTimezoneEnabled, currentTimezone, false).reverse(), [posts]);
 
+    const isTablet = useIsTablet();
+    const galleryIdentifier = 'search-files-location';
+
+    const containerStyle = useMemo(() => {
+        let padding = 0;
+        if (selectedTab === TabTypes.MESSAGES) {
+            padding = posts.length ? 4 : 8;
+        } else {
+            padding = fileInfos.length ? 8 : 0;
+        }
+        return {top: padding};
+    }, [selectedTab, posts, fileInfos]);
+
+    const getChannelName = (id: string) => {
+        return fileChannels.find((c) => c.id === id)?.displayName;
+    };
+
+    const {images: imageAttachments, nonImages: nonImageAttachments} = useImageAttachments(fileInfos, publicLinkEnabled);
+    const filesForGallery = useDerivedValue(() => imageAttachments.concat(nonImageAttachments),
+        [imageAttachments, nonImageAttachments]);
+
+    const orderedFilesForGallery = useDerivedValue(() => (
+        filesForGallery.value.sort((a: FileInfo, b: FileInfo) => {
+            return b.create_at! - a.create_at!;
```

Comment:
Should we default to 0 just in case? (i.e. `(b.create_at || 0) - (a.create_at || 0)`)

Reply:
yes, that is a better option!